### PR TITLE
trust: Implement a 'trust dump' command

### PR DESCRIPTION
**Pull request by stefwalter**
This dumps all the PKCS#11 objects in the internal .p11-kit persistence format.
    
This is part of the trust command and tooling, even though at some point it could go in the p11-kit command. The reason for this is that the code related to the internal .p11-kit objects is in the trust code, and consumed solely by the trust related modules.

### CODE REVIEW

File: trust/list.c [hunk 1]
```diff
@@ -253,6 +253,7 @@ p11_trust_list (int argc,
 	if (!p11_enumerate_ready (&ex, "trust-policy"))
 		exit (1);
 
+	ex.flags |= P11_ENUMERATE_CORRELATE;
```
Comment (ueno) [inline trust/list.c]: Some tests are failing due to this behavior change:
https://travis-ci.org/p11-glue/p11-kit/jobs/196340100
Good to amend this commit with something like ueno/p11-kit@1d9bf793dd2e22853a62988dd736693beae70a77.

File: doc/manual/trust.xml [hunk 2]
```diff
@@ -347,6 +350,42 @@ $ trust extract-compat
 
 </refsect1>
 
+<refsect1 id="trust-list">
```
Comment (ueno) [inline doc/manual/trust.xml]: "trust-dump"

**Comment by kaie** [comment]
Thanks a lot for this patch, it's very helpful!

File: doc/manual/trust.xml (hunk 2 above)
Comment (stefwalter) [inline doc/manual/trust.xml]: Thanks.

File: trust/list.c (hunk 1 above)
Comment (stefwalter) [inline trust/list.c]: Whoops. Obviously I forgot to run 'make check' :O

**Comment by stefwalter** [comment]
Thanks @kaie ... I've amended this patch and force pushed.

**Pull request merged**